core/lib/rom/schema.rb

```diff
@@ -138,7 +138,13 @@ def self.define(name, attributes: EMPTY_ARRAY, attr_class: Attribute, **options)
 
     # @api private
     def self.attributes(attributes, attr_class)
-      attributes.map { |type| attr_class.new(type) }
+      attributes.map do |attr|
+        if attr.is_a?(Hash)
```

Comment:
If I'm not wrong, if we want to be backward compatible with plugins, I can't just change here to use always a Hash. Instead, I'm allowing the type information to be still just the type itself.